From 4d323d561b99132cc8f90f908513c552e6a66240 Mon Sep 17 00:00:00 2001
From: Dan Pasanen <dan.pasanen@gmail.com>
Date: Thu, 15 Mar 2018 20:42:36 -0500
Subject: [PATCH 2/3] PrivacyGuard: skip icon lookup for OP_NONE

* OP 0 (zero) is OP_COARSE_LOCATION. We only want to
  skip a lookup when the OP is OP_NONE (-1).

Change-Id: I64366aff1aa90bb71c2b99dae5485b0262860107
---
 src/com/android/settings/applications/AppOpsDetails.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/applications/AppOpsDetails.java b/src/com/android/settings/applications/AppOpsDetails.java
index 6b4a785..220bdff 100644
--- a/src/com/android/settings/applications/AppOpsDetails.java
+++ b/src/com/android/settings/applications/AppOpsDetails.java
@@ -195,14 +195,14 @@ public class AppOpsDetails extends SettingsPreferenceFragment {
                     mPackageInfo.applicationInfo.uid, mPackageInfo.packageName, true);
             for (final AppOpsState.AppOpEntry entry : entries) {
                 String perm = null;
-                int op = 0;
+                int op = -1;
                 // Find the first permission with a known name
                 for (int i = 0; i < entry.getNumOpEntry() && perm == null; i++) {
                     op = entry.getOpEntry(i).getOp();
                     perm = AppOpsManager.opToPermission(op);
                 }
                 Drawable icon = getIconByPermission(perm);
-                if (icon == null && op != 0 && OP_ICONS.containsKey(op)) {
+                if (icon == null && op != -1 && OP_ICONS.containsKey(op)) {
                     icon = getActivity().getDrawable(OP_ICONS.get(op));
                 }
                 if (icon == null) {
-- 
2.7.4

